internal/datasources/service/service.go

```diff
@@ -262,11 +262,61 @@ func (d *dataSourceService) Update(
 	panic("implement me")
 }
 
-// nolint:revive // there is a TODO
+// Delete deletes a data source in the given project.
 func (d *dataSourceService) Delete(
 	ctx context.Context, id uuid.UUID, project uuid.UUID, opts *Options) error {
-	//TODO implement me
-	panic("implement me")
+	stx, err := d.txBuilder(d, opts)
+	if err != nil {
+		return fmt.Errorf("failed to start transaction: %w", err)
+	}
+	defer func(stx serviceTX) {
+		err := stx.Rollback()
+		if err != nil {
+			fmt.Printf("failed to rollback transaction: %v", err)
```

Comment:
let's not use `fmt.Printf`, you can build a zerologger from the `context.Context` instance.

... Aaaaaand I just realized I missed this in another review. @teodor-yanev can you submit a fix for the `Create` function?